rootex/framework/entity.cpp

```diff
@@ -175,6 +175,7 @@ bool Entity::setScriptJSON(const JSON::json& script)
 		if (OS::IsExists(script["path"]))
 		{
 			m_Script.reset(new Script(script));
+			m_Script->setup(this);
```

Comment:
Why is this needed here?

Reply:
Initially Exception was being thrown on running a scene after adding child scene from file, because for the scenes added from file, `setup` was not being called.